### PR TITLE
rec: log only one line per protocol for listening sockets,

### PR DESCRIPTION
**Pull request by omoerbeek**
Instead of addresses * threads * protocol lines

The recursor startup is too chatty, it's easy to miss important warnings or errors. The socket creation code is the biggest offender (it logs per thread and per listening address for both UDP and TCP), so reduce the amount of lines to just two.

Also stop doing old-style logging for code that is changed and unlikely to be backported, it has been declared unsupported.

The log lines now look like:
```
Dec 09 09:30:01 msg="Listening for queries" subsystem="config" level="0" prio="Info" tid="0" ts="1733733001.553" addresses="0.0.0.0:5301 [::]:5301" proto="UDP" reuseport="1" socketInstances="4"
Dec 09 09:30:01 msg="Listening for queries" subsystem="config" level="0" prio="Info" tid="0" ts="1733733001.553" addresses="0.0.0.0:5301 [::]:5301" protocol="TCP" reuseport="1" socketInstances="1"
```

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Review by chbruyand — APPROVED** [review]

File: pdns/recursordist/rec-main.cc [hunk 1]
```diff
@@ -1891,31 +1891,31 @@ static unsigned int initDistribution(Logr::log_t log)
       for (unsigned int i = 0; i < RecThreadInfo::numDistributors(); i++, threadNum++) {
         auto& info = RecThreadInfo::info(threadNum);
         auto& deferredAdds = info.getDeferredAdds();
-        count += makeUDPServerSockets(deferredAdds, log);
+        count += makeUDPServerSockets(deferredAdds, log, i == RecThreadInfo::numDistributors() - 1, RecThreadInfo::numDistributors());
```
Comment (Habbie) [inline pdns/recursordist/rec-main.cc:1894]: would `i == 0` perhaps be more readable?

Comment (Habbie) [reply pdns/recursordist/rec-main.cc:1894]: it's also lightly confusing that you pass an instance count but the called function does not use it (except for logging), but I see how that would be a bigger rework.

Maybe logging just below this for is cleaner?

**Review by Habbie — APPROVED** [review]
this is not wrong, but i left a question

File: pdns/recursordist/rec-main.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/rec-main.cc:1894]: My idea was to log about socket creation after all has been done, hence the end condition. Let me take a look into logging after the creation.

Comment (Habbie) [reply pdns/recursordist/rec-main.cc:1894]: > hence the end condition.

That did occur to me :)

Comment (Habbie) [reply pdns/recursordist/rec-main.cc:1894]: right. maybe a comment somewhere that explains that the argument is not entirely functional then

**Pull request merged**